src/base/QXmppMessage.cpp

```diff
@@ -236,6 +237,32 @@ void QXmppMessage::setBody(const QString &body)
     d->body = body;
 }
 
+///
+/// Returns an unencrypted string used to be presented to users if the message
+/// could not be decrypted.
+///
+/// \return the end-to-end encryption fallback body
+///
+/// \since QXmpp 1.5
+///
+QString QXmppMessage::e2eeFallbackBody() const
+{
+    return d->e2eeFallbackBody;
+}
+
+///
+/// Sets an unencrypted string used to be presented to users if the message
+/// could not be decrypted.
```

Comment:
The QStrings are always unencrypted. How about this?

Sets an additional message body that is unlike the normal body not encrypted. This way you can have a fallback body for users not supporting the used end-to-end-encryption.